packages/database/src/migrations/20210210011723-UpdateMedicinesAndConsumablesOverlaysFiji-modifies-data.js

```diff
@@ -0,0 +1,637 @@
+  },
+];
+
+const FIJI_VACCINES_OVERLAYS = [
+  {
+    code: 'FijiBCSC157',
+    name: 'DPT-Hib-HBV 1 dose ',
+  },
+  {
+    code: 'FijiBCSC158',
+    name: 'HBV Infant 10mcg (1 dose)',
+  },
+  {
+    code: 'FijiBCSC159',
+    name: 'HBV Adult 20mcg/dose (10 doses)',
+  },
+  {
+    code: 'FijiBCSC161',
+    name: 'Polio Vaccine 10 doses',
+  },
+];
+
+const CONSUMABLE_OVERLAY_IDS = [
+  '64', // BCD59'
+  '65', // BCD60'
+  '66', // BCD68'
+  '67', // BCD70'
+  '68', // BCD71'
+  '69', // BCD72'
+  '70', // BCD73'
+  '71', // BCD74'
+  '72', // BCD75'
+  '73', // BCD76'
+  '74', // BCD77'
+  '75', // BCD78'
+  '76', // BCD79'
+  '77', // BCD80'
+  '78', // BCD81'
+  '79', // BCD87'
+  '80', // BCD88'
+  '81', // BCD89'
+  '82', // BCD90'
+  '83', // BCD91'
+  '84', // BCD92'
+];
+
+const BASE_FIJI_OVERLAY = {
+  measureBuilderConfig: {},
+  measureBuilder: 'valueForOrgGroup',
+  presentationOptions: {
+    displayType: 'color',
+    measureLevel: 'Facility',
+  },
+  countryCodes: '{FJ}',
+  projectCodes: '{disaster,explore}',
+};
+
+const generateMapOverlayIdFromName = overlayName => {
+  const splittedOverlayNames = overlayName
+    .replace(/[^\w\s]/gi, '') // Retain only alphanumeric characters, underscores and spaces.
+    .trim()
+    .split(' ');
+
+  // remove empty elements
+  return splittedOverlayNames.filter(name => name !== '').join('_');
+};
+
+const getLargestSortOrderOfOverlayInGroup = async (db, groupCode) =>
+  db.runSql(`
+    select MAX(map_overlay_group_relation.sort_order) as "largestSortOrder"
+    from "mapOverlay"
+    inner join map_overlay_group_relation on "mapOverlay".id = map_overlay_group_relation.child_id
+    inner join map_overlay_group on map_overlay_group.id = map_overlay_group_relation.map_overlay_group_id
+    where map_overlay_group.code = '${groupCode}'
+    group by map_overlay_group.code
+`);
+
+/**
+ * We receive the dbmigrate dependency from dbmigrate initially.
+ * This enables us to not have to rely on NODE_PATH.
+ */
+exports.setup = function (options, seedLink) {
+  dbm = options.dbmigrate;
+  type = dbm.dataType;
+  seed = seedLink;
+};
+
+const updateMapOverlaySortOrder = async (db, sortOrder, childId, groupId) => {
+  await db.runSql(`
+    UPDATE map_overlay_group_relation
+    SET sort_order = ${sortOrder}
+    WHERE child_id = '${childId}'
+    AND map_overlay_group_id = '${groupId}'
+`);
+};
+
+/**
+ * Shift the Consumable overlays by increasing their sort orders so that we can put new Medicine Overlays in between.
+ * Order should be: Medicine Overlays -> Consumable Overlays
+ * @param {*} db
+ */
+const shiftConsumablesOverlayOrders = async db => {
+  const medicinesAndConsumablesGroupId = await codeToId(
+    db,
+    'map_overlay_group',
+    MEDICINES_AND_CONSUMABLES_OVERLAY_GROUP_CODE,
+  );
+
+  const consumableOverlayStartSortOrder =
+    LAST_MEDICINE_OVERLAY_SORT_ORDER + FIJI_MEDICINES_OVERLAYS.length;
+
+  for (let i = 0; i < CONSUMABLE_OVERLAY_IDS.length; i++) {
```

Comment:
Really nice to see this `for()` syntax, which reminds me of some good time with my old friend Java.